### ISSUE TITLE
UnicodeDecodeError when using the ids_for method

### ISSUE THREAD

**Issue by bitiuscal** [csparpa/pyowm#208]
Hello,

When using the "ids_for" method to get an ID for a city, an error is thrown.
Before that is worked. It worked with Python 3.5.2. First I thought that the error is caused by newly installed Python 3.6.1, but is not. Or that is because I upgraded PyCharm from 2016.3.2 to 2017.1.3.

The error is:
**Traceback (most recent call last):
  File "D:/#_backup/OneDrive/work_python/study/02-xx/02-15a.py", line 70, in <module>
    main(v_API_key)
  File "D:/#_backup/OneDrive/work_python/study/02-xx/02-15a.py", line 19, in main
    print(v_loc01.ids_for(v_loc01name))
  File "C:\python\python353\lib\site-packages\pyowm\webapi25\cityidregistry.py", line 76, in ids_for
    splits = self._filter_matching_lines(city_name, country, matching)
  File "C:\python\python353\lib\site-packages\pyowm\webapi25\cityidregistry.py", line 151, in _filter_matching_lines
    lines = [l.strip() for l in self._get_lines(filename)]
  File "C:\python\python353\lib\site-packages\pyowm\webapi25\cityidregistry.py", line 199, in _get_lines
    lines = fh.readlines()
  File "C:\python\python353\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 6507: character maps to <undefined>**

![2017-08-17_113424 python](https://user-images.githubusercontent.com/31094786/29405015-5926dc38-8345-11e7-8df8-3c8180df4b70.png)

**Comment by csparpa** [comment]
Hi! Thanks for reporting
What location were you trying to search the id for?

**Comment by bitiuscal** [comment]
I tried my own town. But also for London is failing, as you may see in the screen capture (v_loc01name = "London").

**Comment by csparpa** [comment]
Hello @bitiuscal unfortunately I am not able to reproduce the error: I can successfully run your code both on command line interpreter (Python 3.5.2) and PyCharm Community Edition 2017.2.1

Are you running the latest PyOWM version (2.7.1) ???

Please can you create a virtualenv and run this code inside it:

```python
# File: run.py

import pyowm

def main(ip_APIkey):
    print('--start--')
    loc = "London"
    o = pyowm.OWM(ip_APIkey)
    reg = o.city_id_registry()
    print(reg)
    print(reg.ids_for(loc))
    print(reg.locations_for(loc))

if __name__ == '__main__':
    main('prova')
```


then check of the error happens?
Do like this:

```shell
$> sudo pip pinstall virtualenv
$> cd && mkdir testbug && cd testbug
$> virtualenv -p python3.5 .
$> source bin/activate
$> pip install pyowm==2.7.1
$> # create a file named 'run.py' and paste the code into it, then...
$> python run.py
```

your output should be:

```
--start--
<pyowm.webapi25.cityidregistry.CityIDRegistry - filepath_regex=cityids/%03d-%03d.txt.gz>
[(2643743, 'London', 'GB'), (4119617, 'London', 'US'), (4298960, 'London', 'US'), (4517009, 'London', 'US'), (5056033, 'London', 'US'), (5367815, 'London', 'US'), (6058560, 'London', 'CA')]
[<pyowm.webapi25.location.Location - id=2643743, name=London, lon=-0.12574, lat=51.50853>, <pyowm.webapi25.location.Location - id=4119617, name=London, lon=-93.25296, lat=35.328972>, <pyowm.webapi25.location.Location - id=4298960, name=London, lon=-84.08326, lat=37.128979>, <pyowm.webapi25.location.Location - id=4517009, name=London, lon=-83.44825, lat=39.886452>, <pyowm.webapi25.location.Location - id=5056033, name=London, lon=-95.234978, lat=40.445>, <pyowm.webapi25.location.Location - id=5367815, name=London, lon=-119.443176, lat=36.476059>, <pyowm.webapi25.location.Location - id=6058560, name=London, lon=-81.23304, lat=42.983391>]
```

Please let me know the outcome!

**Comment by csparpa** [comment]
@bitiuscal ping..

**Comment by bitiuscal** [comment]
Sorry, I didn't have time to respond.
I don't know how the virtualenv work, so I don't know how to create it. I heard about virtualenv, but never had the time to study.
The pyowm is 2.7.1.

**Comment by csparpa** [comment]
Please @bitiuscal don't drop this issue as other users (#210) faced it!

Virtualenv is a tool that lets you create one-time usage python environments that are isolated from your underlying operating system so we can test the PyOWM installation without any interference by the python libraries that you might have already installed on your operating system before.

Install Virtualenv with:

    $ pip install virtualenv

Then create a Virtualenv with:

    $ cd && mkdir myvirtenv && cd myvirtenv
    $ virtualenv .
    $ source bin/activate

Now you've created a Virtualenv inside the "myvirtenv" subfolder of your homefolder, and that Virtualenv contains the default python interpreter version that your operating system has (might be 3.x or 2.x)

Whenever you're done with the Virtualenv you need to deactivate it:

    $ (myvirtenv) deactivate

And then you can also throw away the "myvirtenv" subfolder

**Comment by csparpa** [comment]
This looks like the same issue as #210 that was fixed with #213
